### PR TITLE
Comparison predicate handling for `!=` 

### LINKED ISSUE #5281: Missing comparison predicate handling for `!=` in Matcher
## How to reproduce the behaviour
```
import spacy
nlp = spacy.load("en_core_web_sm")
matcher = Matcher(nlp.vocab)
pattern = [{"LENGTH": {"!=": 2}}]
matcher.add("LENGTH_COMPARE", [pattern])
doc = Doc(nlp.vocab, words=["a", "aa", "aaa"])
matches = matcher(doc)
assert len(matches) == 2
```

## Your Environment

* **spaCy version:** 2.2.4
* **Platform:** Linux-4.19.0-6-amd64-x86_64-with-glibc2.28
* **Python version:** 3.8.1

### PR DESCRIPTION
**Pull request by paoloq**
## Description

Fix issue #5281.

### Types of change

Bug fix.

## Checklist

- [x] I have submitted the spaCy Contributor Agreement.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy/tests/matcher/test_matcher_api.py [hunk 1]
```diff
@@ -265,14 +265,25 @@ def test_matcher_regex_shape(en_vocab):
...
+@pytest.mark.parametrize(
+    "cmp, ngood, bad", 
+    [
+        ("==", 1, "a aaa"),
+        ("!=", 2, "aa"),
+        (">=", 2, "a"),
+        ("<=", 2, "aaa"),
+        (">", 1, "a aa"),
+        ("<", 1, "aa aaa")
+    ]
```
Comment (svlandeg) [inline spacy/tests/matcher/test_matcher_api.py:277]: I don't think you need `ngood` here: it's always just `3 - length(bad)`, right? Can `bad` be a list (e.g. `[a, aa]`) to make its meaning a bit more clear (and remove the `split()`)

Comment (svlandeg) [reply spacy/tests/matcher/test_matcher_api.py:277]: I do like how you've generalized this unit test, but it took me a second to understand what was going on :-)

**Comment by svlandeg** [comment]
Thanks for the PR, I think this would be a good additional feature to have!

File: spacy/tests/matcher/test_matcher_api.py (hunk 1 above)
Comment (paoloq) [reply spacy/tests/matcher/test_matcher_api.py:277]: Thank you for your feedback, you're definitely right on the optimisation and the ugliness!
I'll fix the test with your hints.

**Pull request merged**